src/runner.rs

```diff
@@ -46,79 +90,147 @@ pub async fn run_test_command(
                     codeowners_path.clone(),
                 )
             })
-            .collect::<anyhow::Result<Vec<FileSet>>>()
-        {
-            Ok(file_sets) => file_sets,
-            Err(e) => {
-                log::error!("Error scanning file sets: {}", e);
-                return Ok(RunResult {
-                    exit_code: result.code().unwrap_or(EXIT_FAILURE),
-                    failures,
-                });
-            }
-        };
-        for file_set in &file_sets {
-            for file in &file_set.files {
-                log::info!("Checking file: {}", file.original_path);
-                let metadata = match metadata(&file.original_path) {
-                    Ok(metadata) => metadata,
-                    Err(e) => {
-                        log::warn!("Error getting metadata: {}", e);
-                        continue;
-                    }
-                };
-                let time = match metadata.modified() {
-                    Ok(time) => time,
-                    Err(e) => {
-                        log::warn!("Error getting modified time: {}", e);
-                        continue;
-                    }
-                };
-                // skip files that were last modified before the test started
+            .collect::<anyhow::Result<Vec<FileSet>>>()?;
+    }
+
+    Ok((file_sets, file_counter))
+}
+
+pub async fn get_failures(
+    file_sets: Vec<FileSet>,
```

Comment:
`file_sets` is only borrowed below, so we can accept a slice here:
```suggestion
    file_sets: &[FileSet],
```